### PR TITLE
fix(masthead-v2): L0 v2.1 Design QA updates

### PR DESCRIPTION
**Pull request by jkaeser**
### Related Ticket(s)

https://jsw.ibm.com/browse/ADCMS-3025

### Description

Addresses design QA feedback from initial L0 v2.1 development work. See https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10066#issuecomment-1448953115.

### Changelog

**Changed**

- Adjusts layout below large breakpoint for Simple layout megamenus with featured areas.
- Prevents Heading Level 1s from appearing in Simple layout megamenus with featured areas.
- Adjusts hover states for mobile back and view all buttons.
- Moves view all CTAs to their proper contextual locations in data model.
- Fixes issue where the `DDSLeftNavigation` would apply incorrect layout rules if only one sub-element was placed inside it.

### To Review

Visit [the deploy preview](https://ibmdotcom-webcomponents.s3.us-east.cloud-object-storage.appdomain.cloud/deploy-previews/10190/iframe.html?args=&knob-platform%20name%20(platform)=&knob-show%20the%20profile%20functionality%20(has-profile)=true&knob-show%20the%20search%20functionality%20(has-search)=true&knob-search%20placeholder%20(searchPlaceholder)=Search%20all%20of%20IBM&knob-selected%20menu%20item%20(selected-menu-item)=Consulting%20&%20Services=&knob-The%20user%20authenticated%20status%20(user-status)=Unauthenticated&knob-use%20mock%20nav%20data%20(use-mock)=true&id=components-masthead--with-v-2-data&viewMode=story) and verify:
- "Featured" layout megamenus break to a single column below the 800px breakpoint
- "Featured" layout megamenus do not display Heading Level 1 in megamenus or mobile menu sections.
- "View all" CTAs stick to the bottom of megamenu columns.
- "Back" and "View All" buttons in the mobile menu have the same hover states as their tablet/desktop counterparts.

### CODE REVIEW

**Comment by jkaeser** [comment]
Found a bug with the megamenu right columns' heights in certain circumstances. Please hold off on reviews until fixed.

**Comment by jkaeser** [comment]
> Found a bug with the megamenu right columns' heights in certain circumstances. Please hold off on reviews until fixed.

This has been resolved. Ready for review!

File: packages/styles/scss/components/masthead/_masthead-megamenu.scss [hunk 1]
```diff
@@ -123,13 +132,109 @@
...
+          @include carbon--breakpoint-down(lg) {
+            --#{$dds-prefix}-view-all-padding: #{$layout-01};
+            --#{$dds-prefix}-masthead-max-height: initial;
+            --#{$dds-prefix}-masthead-overflow: initial;
+
+            width: 100%;
+            max-width: initial;
+            margin-inline: initial;
+            padding-inline: $layout-03;
+            border-bottom: 1px solid $ui-03;
```
Comment (proeung) [inline packages/styles/scss/components/masthead/_masthead-megamenu.scss:192]:
```suggestion
            border-bottom: carbon--rem(1px) solid $ui-03;
```

Nitpicking: Let's keep this to `rem` and consistent with how it's being used here (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10190/files#diff-56ec093065bb965a14aa4a46ec7c41d439780a22109f50f4ebed646596f10fb3R287)

File: packages/styles/scss/components/masthead/_masthead-megamenu.scss [hunk 2]
```diff
@@ -123,13 +132,109 @@
...
+        }
+
+        ::slotted([part*='view-all']) {
+          --#{$dds-prefix}-view-all-padding: #{$layout-04};
+
+          position: sticky;
+          z-index: 1;
+          inset-block-end: 0;
+          background-color: $ui-background;
+          border-top: 1px solid $ui-03;
```
Comment (proeung) [inline packages/styles/scss/components/masthead/_masthead-megamenu.scss:232]: Same comment - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/10190/files#r1134581317

**Comment by jkaeser** [comment]
@oliviaflory Thanks for the review! I pushed fixes for the issues you caught, so please take another look when you have a chance.

**Comment by jwitkowski79** [comment]
@jkaeser As per @proeung I did a quick review and the changes that @oliviaflory requested look good! Approval from me

**Pull request merged**